src/lib.rs

```diff
@@ -1205,12 +1190,13 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
                 // new key, so keep incrementing the key stream state and generating keys until we
                 // find one that is new.
                 loop {
+                    let index = model.sending_accounts.index();
                     let user_key = model
                         .user_key_stream
-                        .derive_user_key_pair(&self.key_state.user.to_le_bytes());
-                    self.key_state.user += 1;
+                        .derive_user_key_pair(&index.to_le_bytes());
+                    model.sending_accounts.increment_index();
```

Comment:
Could maybe make this cleaner by having a `next_index` function on `Assets` which both returns and increments the index

Reply:
Sounds good, assuming you meant `Accounts`?

Reply:
Yes, my bad